File: 3024_Type_Of_Triangle.cpp
class Solution {
public:
    string triangleType(vector<int>& z) {
        sort(z.begin(), z.end());
        if (z[0] + z[1] <= z[2]) return "none";
        return z[0] == z[2] ? "equilateral" : (z[0] == z[1] || z[1] == z[2]) ? "isosceles" : "scalene";
    }
};

File: 3024_Type_Of_Triangle.py
class Solution(object):
    def triangleType(self, z):
        z.sort()
        return "none" if z[0] + z[1] <= z[2] else "equilateral" if z[0] == z[2] else "isosceles" if z[0] == z[1] or z[1] == z[2] else "scalene"
